test/gurka/test_shortcut.cc

```diff
@@ -42,6 +42,37 @@ TEST(Shortcuts, CreateValid) {
   EXPECT_EQ(opp_shortcut_edged->speed(), 90);
 }
 
+TEST(Shortcuts, LoopWithoutShortcut) {
+  constexpr double gridsize = 50;
+
+  const std::string ascii_map = R"(
+      A--B
+      |  |
+      |  |
+      C--D
+  )";
+  const gurka::ways ways = {
+      {"AB",
+       {{"highway", "primary"}}},
```

Comment:
The test is simple and effective, thanks. However, I don’t think this would fail without your PR. I think the edges need more common attributes for them to form a shortcut IIRC.

Reply:
I've tried and it runs into infinite loop without my PR, because GetShortcut try to find a shortcut and run along the edges in the loop. It doesn't fail, but run for ever without my PR.

Reply:
Ok perfect, I was pretty sure I was pulling my hair out a few times when it didn't produce shortcuts for some reason when I didn't include any other properties like `name` or so. Thanks for the feedback, LGTM.